python/amici/petab_import.py

```diff
@@ -574,35 +574,127 @@ def import_model_sbml(
 import_model = import_model_sbml
 
 
-def get_observation_model(observable_df: pd.DataFrame
-                          ) -> Tuple[Dict[str, Dict[str, str]],
-                                     Dict[str, str],
-                                     Dict[str, Union[str, float]]]:
+def has_timepoint_specific_measurement_overrides(measurement_df: pd.DataFrame):
+
+    def floatable(x: str):
+        try:
+            [float(x) for x in x.split(';')]
+            return True
+        except ValueError:
+            return False
+
+    nan_pat = r'^[nN]a[nN]$'
+
+    has_timepoint_noise_overrides, has_timepoint_observable_overrides = [
+        len([x for x in measurement_df[field].unique()
+             if isinstance(x, str) and not re.match(nan_pat, x)
+             and not floatable(x)]) > 0
+        if field in measurement_df else False
+        for field in [petab.NOISE_PARAMETERS, petab.OBSERVABLE_PARAMETERS]
+    ]
+    return has_timepoint_noise_overrides or has_timepoint_observable_overrides
+
+
+def flatten_timepoint_specific_overrides(problem: petab.Problem):
```

Comment:
Looks like a better implementation than https://petab.readthedocs.io/en/stable/build/_autosummary/petab.core.html#petab.core.flatten_timepoint_specific_output_overrides. Maybe we can update the function there afterwards. There is already some test case.